File: app/src/main/java/com/ac/demo/Book.java
package com.ac.demo;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;

/**
 * Created by anna on 2017/11/29.
 */

@Entity(tableName = "book", foreignKeys = @ForeignKey(entity = User.class,
        parentColumns = "id",
        childColumns = "user_id"), indices = {@Index(value ={"user_id","book_id"})})
class Book {
    @PrimaryKey
    @ColumnInfo(name = "book_id")
    public int bookId;

    public String title;

    @ColumnInfo(name = "user_id")
    public int userId;

    public int getBookId() {
        return bookId;
    }

    public void setBookId(int bookId) {
        this.bookId = bookId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }
}

File: simple/src/main/java/com/kotlin/sample/persistence/Book.kt
package com.kotlin.sample.persistence

import android.arch.persistence.room.*
import com.example.android.observability.persistence.User

/**
 * Created by anna on 2017/12/13.
 */
@Entity(tableName = "book", foreignKeys = arrayOf(ForeignKey(entity = User::class,
        parentColumns = arrayOf("id"), childColumns = arrayOf("user_id"))),
        indices = arrayOf(Index(value = *arrayOf("user_id", "book_id"))))
data class Book(@PrimaryKey
                @ColumnInfo(name = "book_id")
                var bookId: Int = 0,
                var title: String,
                @ColumnInfo(name = "user_id")
                var userId: Int = 0
)